### PR TITLE
potential workaround for M FileObserver bug

### PR DESCRIPTION
**Pull request by cgoodroe**

### CODE REVIEW

File: bugreport-library/src/main/java/com/doubleencore/bugreport/internal/ScreenshotObserver.java [hunk 1]
```diff
@@ -46,14 +56,44 @@ public void onEvent(int event, String path) {
         return true;
     }
 
+    @TargetApi(Build.VERSION_CODES.M)
+    public static boolean enableObserverMushroom(final FileAlterationListener listener) {
```
Comment (chris-mitchell) [inline bugreport-library/src/main/java/com/doubleencore/bugreport/internal/ScreenshotObserver.java]: Can we rename to Marshmellow?

File: bugreport-library/src/main/java/com/doubleencore/bugreport/BugReport.java [hunk 2]
```diff
@@ -35,7 +36,11 @@ public static void executeCollection(@NonNull Activity activity) {
 
     public static void enableObserver(@NonNull Activity activity) {
         if (checkPermissions(activity, BugReport.ENABLE_OBSERVER)) {
-            ScreenshotObserver.enableObserver(BugReportInternal.getInstance());
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (chris-mitchell) [inline bugreport-library/src/main/java/com/doubleencore/bugreport/BugReport.java]: Can we move the if / else into `ScreenshotObserver.enableObserver` ? This class doesn't need to be aware of the differences

File: bugreport-library/src/main/java/com/doubleencore/bugreport/internal/ScreenshotObserver.java [hunk 3]
```diff
@@ -46,14 +56,44 @@ public void onEvent(int event, String path) {
...
+        mFileAlterationListener = listener;
+        if (mFileAlterationObserver == null) {
+            try {
+                mFileAlterationObserver = new FileAlterationObserver(getScreenshotDirectory());
+                mFileAlterationObserver.addListener(mFileAlterationListener);
+                mFileAlterationObserver.initialize();
+                mFileAlterationMonitor = new FileAlterationMonitor(1000L);
```
Comment (chris-mitchell) [inline bugreport-library/src/main/java/com/doubleencore/bugreport/internal/ScreenshotObserver.java]: Could we use TimeUnit to get 1 second into 1000 ?

**Comment by cgoodroe** [comment]
@chris-mitchell updated with less mushrooms :(

**Comment by cgoodroe** [comment]
@chris-mitchell updated

**Comment by cgoodroe** [comment]
@chris-mitchell hoping this is what you wanted :)

**Comment by chris-mitchell** [comment]
Awesome! Thanks!

**Pull request merged**